rai/secure.cpp

```diff
@@ -1990,7 +1990,7 @@ class rollback_visitor : public rai::block_visitor
 		rai::account_info info;
 		auto error (ledger.store.account_get (transaction, destination_account, info));
 		assert (!error);
-		ledger.store.representation_add (transaction, ledger.representative (transaction, hash), 0 - amount);
+		ledger.store.representation_add (transaction, representative, 0 - amount);
```

Comment:
This change basically implies that

`ledger.representative (transaction, block_a.hashables.previous)`

is the same as

`ledger.representative (transaction, block_a.hash ())`

Is that right?

Reply:
Yeah I don't think this change is correct.

Reply:
Yeah I don't think this change is correct.

Reply:
rep is chosen by iterating block's previous until open or change block is found, so I think the change is correct.